platform/viewer/src/studylist/StudyListRoute.js

```diff
@@ -172,33 +224,143 @@ function StudyListRoute(props) {
     );
   }
 
-  function handleSort(fieldName) {
-    let sortFieldName = fieldName;
-    let sortDirection = 'asc';
-
-    if (fieldName === sort.fieldName) {
-      if (sort.direction === 'asc') {
-        sortDirection = 'desc';
-      } else {
-        sortFieldName = null;
-        sortDirection = null;
+  const handleSort = useCallback(
+    fieldName => {
+      let sortFieldName = fieldName;
+      let sortDirection = 'asc';
+
+      if (fieldName === sort.fieldName) {
+        if (sort.direction === 'asc') {
+          sortDirection = 'desc';
+        } else {
+          sortFieldName = null;
+          sortDirection = null;
+        }
       }
-    }
 
-    setSort({
-      fieldName: sortFieldName,
-      direction: sortDirection,
-    });
-  }
+      const newProps = {
+        sort: {
+          fieldName: sortFieldName,
+          direction: sortDirection,
+        },
+      };
+      const newStudyListSessionProps = mergeStudyListSessionProps(
+        studyListSessionProps,
+        newProps
+      );
+      fetchStudies(newStudyListSessionProps, false);
+    },
+    [studyListSessionProps]
+  );
 
-  function handleFilterChange(fieldName, value) {
-    setFilterValues(state => {
-      return {
-        ...state,
-        [fieldName]: value,
+  const handleFilterChange = useCallback(
+    newFilterObj => {
+      const newProps = {
+        filters: {
+          ...filterValues,
+          ...newFilterObj,
+        },
+        pageNumber: DEFAULT_PAGE_NUMBER,
       };
-    });
-  }
+
+      const newStudyListSessionProps = mergeStudyListSessionProps(
+        studyListSessionProps,
+        newProps
+      );
+      fetchStudies(newStudyListSessionProps, false);
+    },
+    [studyListSessionProps]
+  );
+
+  const clearFilters = useCallback(() => {
+    const newProps = {
+      filters: DEFAULT_FILTERS,
+      pageNumber: DEFAULT_PAGE_NUMBER,
+    };
+
+    const newStudyListSessionProps = mergeStudyListSessionProps(
+      studyListSessionProps,
+      newProps
+    );
+    fetchStudies(newStudyListSessionProps);
+  }, [studyListSessionProps]);
+
+  const setPageNumberSessionStorage = useCallback(
+    pageNumber => {
+      const newProps = {
+        pageNumber,
+      };
+
+      const newStudyListSessionProps = mergeStudyListSessionProps(
+        studyListSessionProps,
+        newProps
+      );
+      fetchStudies(newStudyListSessionProps);
+    },
+    [studyListSessionProps]
+  );
+
+  const setRowsPerPageSessionStorage = useCallback(
+    rowsPerPage => {
+      const newProps = {
+        rowsPerPage,
+      };
+
+      const newStudyListSessionProps = mergeStudyListSessionProps(
+        studyListSessionProps,
+        newProps
+      );
+      fetchStudies(newStudyListSessionProps);
+    },
+    [studyListSessionProps]
+  );
```

Comment:
Some repeat code here for setting these. Would it make more sense to shift from individual pieces of state to a "filter + sort" object, and then handle all of the callback/session/updates in once place?

Reply:
yeah we could, but to avoid having component re-render multiple times and considering things might change simultaneously (i.e page+filter, sort+something else...) this implementation ensure unicity and scalability. We can loop through it for any change if you desire to